rir/src/compiler/opt/cleanup.h

```diff
@@ -1,13 +1,19 @@
 #ifndef PIR_CLEANUP_H
 #define PIR_CLEANUP_H
 
+#include "../translations/rir_compiler.h"
+
 namespace rir {
 namespace pir {
 
 class Closure;
-class Cleanup {
+class Cleanup : public PirTranslator {
   public:
-    static void apply(Closure* function);
+    Cleanup()
+        : PirTranslator("cleanup") {};
+
+  protected:
+    void applyTranslation(Closure* function);
```

Comment:
please always mark implementations of `virtual` functions with `override` or `final`